### DISCUSSION TITLE
[Feature idea] Option for evenly spaced desktop item margins

### DISCUSSION THREAD

**Discussion by isf63** [lxqt/lxqt#2540] [Ideas]
In PCManFM-Qt's Desktop Preferences, the user can set 'Minimum item margins' which specify how much spacing surrounds each desktop item.

Windows and KDE by default have automatic desktop item margins that evenly space out the icons, using a similar method to:
`num_items = floor(screen_width / item_width)`
`item_margin_width = max(minimum_item_margin_width,  (screen_width - (num_items * item_width)) / (num_items + 1) / 2`
`item_margin_height = ...` same as width, but using vertical dimensions.

With manual margins if you place icons on the edges of the screen, as is common, you can see varying spacing. An automatic option would solve this.

**Comment by tsujan** [comment]
> With manual margins if you place icons on the edges of the screen, as is common, you can see varying spacing.

There's no varying spacing: only one horizontal spacing and one vertical spacing exist. The vertical spacing is automatically adjusted to avoid an empty area at the bottom when columns are full.

**Reply by isf63** [reply]
I meant between screen edges. Windows and KDE by default space margins evenly, as in `|--x--x--x--x--|`, where `|` is the screen edge, `-` is the margin, and `x` is the item. It's currently achievable sort of in LXQt by setting specific minimum item margins. As it is common to place desktop items along the edges, and for easy use, I think having an option for this would be beneficial.

**Reply by tsujan** [reply]
The equal top and bottom margins (default) are always respected. Of course, for various reasons, you might make them unequal.

Also, all item rectangles have the same size, which is the icon size plus three lines of text with the same width. So, if you feel spacings aren't equal vertically, that's because of different icons and/or text lengths.

> as in |--x--x--x--x--|,...

If you mean that an item put at the right edge may not have the same distance from it as an item at the left edge, yes, that depends on the relevant sizes (icon, font, screen....). The distances can be made equal in the same way used for the top and bottom edges, but the code complexity and extra CPU time aren't justified for an app that wants to be as lightweight, manageable and bug-free as possible (neither KDE nor Windows care about these things).

**Reply by isf63** [reply]
>icon size plus three lines of text with the same width

Yes. I wouldn't want to equally space, say 2 and then 3 lines items. The bounding boxes are good as they are now.

>code complexity

Isn't it just a matter of 4 values - screen width/height and item bounding box width/height, and two simple calculations?

Regardless, this can be done by manually inputting specific 'Minimum item margins' I believe. That's what I've been doing to achieve this.

**Reply by tsujan** [reply]
> screen width/height and item bounding box width/height

As mentioned above, the vertical distances ("height") were made equal many years ago. The reason was that items were positioned from top to bottom. There was no practical need to similar calculations for the horizontal right gap (or left gap for RTL, which would complicate them).

> Isn't it just a matter of 4 values

Extra computations/options that are made here and there could have a tangible effect when put together. Only usability might justify them.

> this can be done by manually

I wanted to suggest that. The user can play with the minimum horizontal item margin and the left/right margin of the workspace. But it's just a matter of taste.
